### PR TITLE
Fix LIBBPF_OPTS noise

### PR DESCRIPTION
**Pull request by hhoffstaette**
- add `-Werror=missing-field-initializers` to the CI build config so that accidental use of the wrong macro fails the build
- wrap LIBBPF_OPTS into a pragma-guarded macro to silence `-Wmissing-field-initializers`

Fixes: #3276 

##### Checklist

- [x] Language changes are updated in `man/adoc/bpftrace.adoc`
- [x] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [x] The new behaviour is covered by tests

### CODE REVIEW

**Comment by hhoffstaette** [comment]
So it turns out the first cut works with clang but NOT with gcc because [reasons](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=78657) :angry: 
However after some fiddling I found that this seems to only affect the push/pop pragmas, and removing them allows this to work after all. The only other alternative would have been to duplicate the LIBBPF_OPS definition itself.

**Review by jordalgo — APPROVED** [review]
LGTM, but I'll wait for @viktormalik to review.

File: src/bpffeature.h [hunk 1]
```diff
@@ -54,6 +54,10 @@ public:                                                                        \
 #define DEFINE_PROG_TEST(var, progtype)                                        \
   __DEFINE_PROG_TEST(var, progtype, NULL, std::nullopt)
 
+#define BPFTRACE_LIBBPF_OPTS(TYPE, NAME, ...)                                  \
+  _Pragma("GCC diagnostic ignored \"-Wmissing-field-initializers\"")           \
+      LIBBPF_OPTS(TYPE, NAME, __VA_ARGS__)
```
Comment (danobi) [inline src/bpffeature.h:59]: nit: I think it's better to define this in `src/libbpf/bpf.h`

File: flake.nix [hunk 2]
```diff
@@ -98,6 +98,7 @@
                 # Release flags
                 cmakeFlags = [
                   "-DCMAKE_BUILD_TYPE=Release"
+                  "-DCMAKE_CXX_FLAGS=-Werror=missing-field-initializers"
```
Comment (danobi) [inline flake.nix:101]: So I think the scope is wider than just CI - we'd like anyone building without nix to hit this too, right? If so, probably throw this in here? https://github.com/bpftrace/bpftrace/blob/3e915ae21d32418ac56f4e6174f96918a5926d9e/CMakeLists.txt#L36-L46

**Review by danobi — COMMENTED** [review]
small nits, lgtm otherwise

File: flake.nix (hunk 2 above)
Comment (hhoffstaette) [reply flake.nix:101]: I know *nothing* about nix and only found the cmakeFlags while spelunking. :)
During testing I added -DCMAKE_CXX_FLAGS to verify, but I agree it makes sense to add the -Werror bit to all builds. Will do.

File: src/bpffeature.h (hunk 1 above)
Comment (hhoffstaette) [reply src/bpffeature.h:59]: Fine with me - will do.

**Review by danobi — APPROVED** [review]
thanks for the work on fixing warnings!

**Review by viktormalik — APPROVED** [review]
Nice, thanks!

**Pull request merged**